apis/core/v1beta1/featureflagconfiguration_types.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+	"github.com/open-feature/open-feature-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// FeatureFlagConfigurationSpec defines the desired state of FeatureFlagConfiguration
+type FeatureFlagConfigurationSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// +optional
+	// +nullable
+	ServiceProvider *FeatureFlagServiceProvider `json:"serviceProvider"`
+	// +optional
+	// +nullable
+	SyncProvider *FeatureFlagSyncProvider `json:"syncProvider"`
+	// +optional
+	// +nullable
+	FlagDSpec *FlagDSpec `json:"flagDSpec"`
+	// FeatureFlagSpec is the structured representation of the feature flag specification
+	FeatureFlagSpec FeatureFlagSpec `json:"featureFlagSpec,omitempty"`
+}
+
+type FlagDSpec struct {
+	// +optional
+	Envs []corev1.EnvVar `json:"envs"`
+}
+
+type FeatureFlagSpec struct {
```

Comment:
Just as a counterpoint to creating a new resource @skyerus , I think you could have also gone down this route:

```
type FeatureFlagConfigurationSpec struct {


Styling with Markdown is supported
        // +kubebuilder:deprecatedversion:warning="core.openfeature.dev/v1alpha1 FeatureFlagSpec is deprecated"
       // FeatureFlagSpec string `json:"featureFlagSpec, omityempty"`

	Flags map[string]FlagSpec `json:"flags"`
	// +optional
	// +kubebuilder:validation:Schemaless
	// +kubebuilder:pruning:PreserveUnknownFields
	// +kubebuilder:validation:Type=object
	Evaluators json.RawMessage `json:"$evaluators,omitempty"`
}
```